libs/designsystem/src/lib/components/modal/modal-wrapper/modal-wrapper.component.ts

```diff
@@ -125,6 +132,26 @@ export class ModalWrapperComponent implements Modal, AfterViewInit, OnInit, OnDe
       providers: [{ provide: COMPONENT_PROPS, useValue: this.config.componentProps }],
       parent: this.injector,
     });
+
+    if (this.config.interactWithBackground) {
+      merge(this.ionModalDidPresent, this.viewportResize$)
+        .pipe(takeUntil(this.destroy))
+        .subscribe(() => {
+          // wait for template to render
+          setTimeout(() => {
+            const topPosition =
+              this.windowRef.document.documentElement.offsetHeight -
+              this.elementRef.nativeElement.scrollHeight;
+            const horizontalPosition =
+              (this.windowRef.document.documentElement.offsetWidth -
+                this.elementRef.nativeElement.scrollWidth) /
+              2;
+            this.ionModalElement.style.top = `${topPosition}px`;
+            this.ionModalElement.style.left = `${horizontalPosition}px`;
+            this.ionModalElement.style.right = `${horizontalPosition}px`;
+          });
+        });
+    }
```

Comment:
Consider readability and extract to separate method (same as `this.initializeSizing, this.initializeModalRoute` etc